projects/packages/my-jetpack/_inc/components/product-card/use-pricing-data.ts

```diff
@@ -33,18 +32,21 @@ const parsePricingData = ( pricingForUi: ProductCamelCase[ 'pricingForUi' ] ) =>
 	}
 
 	const {
+		discountPrice,
 		discountPricePerMonth,
+		fullPrice,
 		fullPricePerMonth,
 		currencyCode,
-		isIntroductoryOffer,
 		wpcomProductSlug,
 	} = pricingForUi;
+	const hasDiscount = discountPrice && discountPrice !== fullPrice;
+	const eligibleForIntroDiscount = ! introductoryOffer?.reason;
 	return {
 		wpcomFreeProductSlug,
 		wpcomProductSlug,
 		discountPrice:
 			// Only display discount if site is elgible
-			isIntroductoryOffer && ! introductoryOffer?.reason ? discountPricePerMonth : null,
+			hasDiscount && eligibleForIntroDiscount ? discountPricePerMonth : null,
```

Comment:
Correct me if I am wrong, but I think we can do it this way where the discount isn't shown if the site is not eligible for an intro discount since sales are always for new purchases only 🤔 Let me know if I am wrong or am missing something. There may be an edge case in there that this would not work for

Reply:
Yes I think this is correct. I mean, it sounds right to me! 👍